src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -350,6 +366,184 @@ public String getImageDownloadLink(long imageID) {
+    }
+    return null;
+  }
+
+  /**
+   * Generate the ome.tiff file from a given image id
+   *
+   * @param imageId to define the image for which the ome.tiff should be generated
+   * @return the generate ome.tiff as a file
+   */
+  private File generateOmeTiff(long imageId) {
+    if (!this.isConnected()) {
+      connect();
+    }
+    final int BUFFER_SIZE = 1024 * 1024;
+
+    File generatedTiff = null;
+
+    try {
+      generatedTiff = File.createTempFile("generated_" + imageId + "_",".ome.tiff");
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    try {
+      ExporterPrx exporterService = gateway.getExporterService(securityContext);
+      exporterService.addImage(imageId);
+      long fileLength = exporterService.generateTiff();
+
+      FileOutputStream fileOutputStream = new FileOutputStream(generatedTiff);
+
+      int bytesRead = 0;
+      while (bytesRead < fileLength) {
+        byte[] currentImageSlice = exporterService.read(bytesRead, BUFFER_SIZE);
+        fileOutputStream.write(currentImageSlice);
+        bytesRead += currentImageSlice.length;
+      }
+
+      fileOutputStream.close();
+      exporterService.close();
+
+    } catch (ServerError | DSOutOfServiceException | FileNotFoundException exception) {
+      throw new RuntimeException("Omero could not create the ome tiff for image " + imageId, exception);
+    } catch (IOException ioException) {
+      throw new RuntimeException("Could not write ome.tiff to temporary file for image " + imageId, ioException);
+    }
+    return generatedTiff;
+  }
+
+
+  /**
+   * Attaches a FileAnnotation to the image containing the provided file. This methods also stores the file in the RawFileStore
+   *
+   * @param imageId the image an annotation is attached to
+   * @param file the file that will be contained in the annotation
+   * @return the annotation ID for the attached annotation
+   */
+  private long attachFileAnnotation(long imageId, File file) {
+
+    String fileName = file.getName();
+    String absolutePath = file.getAbsolutePath();
+    String folderPath = absolutePath.substring(0, absolutePath.length() - fileName.length());
+
+    final ChecksumAlgorithm checksumAlgorithm = new ChecksumAlgorithmI();
+    checksumAlgorithm.setValue(omero.rtypes.rstring(ChecksumAlgorithmSHA1160.value));
+
+    OriginalFile originalFile = new OriginalFileI();
+    originalFile.setName(omero.rtypes.rstring(fileName));
+    originalFile.setPath(omero.rtypes.rstring(folderPath));
+    originalFile.setSize(omero.rtypes.rlong(file.length()));
+    originalFile.setHasher(checksumAlgorithm);
+    originalFile.setMimetype(omero.rtypes.rstring("OMETiff"));
+
+    try {
+      DataManagerFacility dataManagerFacility = gateway.getFacility(DataManagerFacility.class);
+      originalFile = (OriginalFile) dataManagerFacility.saveAndReturnObject(securityContext, originalFile);
+
+
+
+      long position = 0;
+      int readLength;
+      final int BYTE_INCREMENT = 262144;
```

Comment:
Is this due to some memory limit? 256 KiB? A comment would be helpful here.

Reply:
We could not figure out what limit this is nor how high we should set it. We took the tutorial code example here because it worked.